sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonSerializer.java

```diff
@@ -29,8 +31,28 @@
         return gson.fromJson(CoreUtils.bomAwareToString(input, null), clazz);
     }
 
+    @Override
+    public <T> T deserializeTree(JsonNode jsonNode, Class<T> clazz) {
+        return gson.fromJson(JsonNodeUtils.toGsonElement(jsonNode), clazz);
+    }
+
     @Override
     public byte[] serialize(Object value) {
         return gson.toJson(value).getBytes(StandardCharsets.UTF_8);
     }
+
+    @Override
+    public byte[] serializeTree(JsonNode jsonNode) {
+        return gson.toJson(JsonNodeUtils.toGsonElement(jsonNode)).getBytes(StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public JsonNode toTree(byte[] input) {
+        return JsonNodeUtils.fromGsonElement(new JsonParser().parse(new String(input, StandardCharsets.UTF_8)));
```

Comment:
We default to `StandardCharsets.UTF_8`, wondering whether we should take charset as input param in case  the over the wire bytes are in a different charset https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type

Reply:
Could use `FluxUtil.bomAwareToString` in this case, though I believe the GSON and Jackson parsers will handle this conversion properly when dealing with a raw byte array. (I believe the issue we were seeing in Azure Core a few months back was due to a naive string conversion before passing into the serializer).